DDF-2644: Cleaned up comments, variables, and javadoc

rzwiefel opened the pull request.

shaundmorris commented:
I verified this fix works on the 2D and 3D maps. @mplachec the only time I couldn't get rid of the features on the map is when I tried to complete the polygon prior to 3 points, so not sure if that is a separate issue. @pklinef or @stustison any ideas on that?

stustison commented:
Yeah that is a separate issue. I believe @mplachec wrote up another ticket for it yesterday.

shaundmorris commented:
Okay I didn't see a ticket so I'll wait on that until I hear from @mplachec on it, just want to make sure we capture it. If there is nothing else @stustison I believe we can merge this in.

mplachec commented:
@shaundmorris There are two issues with the polygons (one on 3d and one on 2d) that I wrote up tickets for:
https://jira.di2e.net/browse/GDIB-970
https://jira.di2e.net/browse/GDIB-967

andrewkfiedler committed:
Update preferences to save when the user goes between map and histogram

djblue committed:
Removed unused less files.

stustison committed:
Fixed map layer preferences bugs

andrewkfiedler committed:
Add back necessary less file to compilation, remove unnecessary function
 - A necessary less file was recently removed from compilation by accident (menu-vertical).
 - User did not need a generateID function as one already exists in Common.  It has been updated to use the latter.

djblue committed:
Several announcement fixes.

- Switched to more idiomatic react animation.
- Added dim higher order component for passing dimensions to wrapped
  componnets.
- Switched to using redux and react-redux
- Updated tests.

djblue committed:
Remove Beta warning.

andrewkfiedler committed:
Add contextual help view, add some help messages to various common views
 - There is now a way to view contextual help.  Clicking the question mark in the title bar will toggle a view that shows help anywhere it has been defined in the html (using the attribute 'data-help').  It figures out what elements are visible in the viewport and only shows help for those elements.
 - Added 'data-help' attributes to a bunch of html templates to start off contextual help.

andrewkfiedler committed:
Make hints have a max width, update hint padding to be consistent, use variable for z-index

andrewkfiedler committed:
Move max width css to correct element for dropdown hints, remove workspace help

andrewkfiedler committed:
Update blend mode for help view
 - Adds blend mode setting for help view.  This lets us make a window to the original color and brightness of the object underneath.  As such, elements with hints won't lose legibility or opacity when going into help mode.
 - Updates dropdown hints to have a brighter border upon hover.

djblue committed:
Added dirty workspace check on page close.

andrewkfiedler committed:
Add new ways to escape help view
 - Users can now leave the help view by clicking on something that isn't a hint, or pressing the escape key.  Other keys are allowed, so that users can print screen, copy hints, etc.

andrewkfiedler committed:
Update announcements to de-duplicate themselves
 - If an announcement that comes in is identical to a previous announcement, the previous announcement gets automatically dismissed.

andrewkfiedler committed:
Add ability to turn off automatic ajax error handling on per request basis
 - Turns off automatic ajax error handling for the ingest requests since those errors are already displayed in their own custom way.

djblue committed:
Update to pass jenkins build.

djblue committed:
Fixed how tests are run for jenkins.

Commit:
added catalog-ui app

Commit:
cleaned up catalog ui app pom

djblue committed:
Updated DELETE endpoint to return valid JSON message.

- This was causing an error to appear on successful archiving.

andrewkfiedler committed:
Refactor theme to be more consistent, reduce number of less variables
 - Updates styles to be more consistent, but also to reduce the necessary maintenance cost.

andrewkfiedler committed:
Update metacard definitions with new defaults, add defensive checks to filtering and summary

andrewkfiedler committed:
Update input theme to be more consistent with the overall theme

andrewkfiedler committed:
Update disabled input theme
 - Contrast with previous theme was too low.

djblue committed:
Added catalog-ui-app as a default app.

andrewkfiedler committed:
Remove / collapse some less variables

andrewkfiedler committed:
Add branding and welcome

andrewkfiedler committed:
Fix issue with saving preferences from event trigger
 - The listenTo function was passing the event, so that was inadvertently being sent as the options for the backbone sync.  This was causing a server error to appear.  Now it will just call save.
 - Updated the backbone cometd extension to be slightly more defensive (in case options aren't passed).

andrewkfiedler committed:
Fix issue with blank query for recent uploads

djblue committed:
Added login on enter key press.

andrewkfiedler committed:
Add autocomplete for search criteria
 - Refer to http://stackoverflow.com/questions/15462991/trigger-autocomplete-without-submitting-a-form for details as to why we need an iframe.

andrewkfiedler committed:
Update padding to be applied to inner container so edit buttons aren't affected

andrewkfiedler committed:
Fix issues with map layer preferences
 - Layers weren't being properly compared, causing layers to be removed and readded when they
shouldn't have been.
 - Switched some references to cid to be id.

andrewkfiedler committed:
Add "read only" capability
 - Users are now restricted from editing attributes called out as read only in the configuration for the UI.
 - Read only attributes are distinguished by both text and background color.

andrewkfiedler committed:
Add query actions availability to results panel
 - The query actions are no longer hidden from the results panel.  Now you can run, delete, duplicate, or stop a query from the results panel.

pklinef committed:
Fix map layer preferences sorting and reset defaults bugs

andrewkfiedler committed:
Remove handlebars template hot loading, fix clustering svg compiling
 - Removed handlebars template hot loading as it was causing issues with manual template compiling.
 - Updated hint.hbs to use "this.hint".  Previously, "hint" wouldn't work because the webpack template loader for handlebars would try to require in the "hint.js" file in the same directory.  There doesn't appear to be an easy way to disable this behavior, so from now on we should stick to prepending property references with "this." in handlebars files.

andrewkfiedler committed:
Remove unused variable from preferences.modal.view

andrewkfiedler committed:
Fix issue with geometry collections
 - There was a usage of _.clone that should have been using the backbone prototype's clone.
 - All WKT geometries will now show up on the map correctly.

andrewkfiedler committed:
Points handlebars require to correct location

pklinef committed:
Allow multi geometries to overlap when parsing

andrewkfiedler committed:
Remove unused bootswatch theme

andrewkfiedler committed:
Overwrite bootstrap hover color

djblue committed:
Workspace security updates:

- Allow workspace owner to be specified at creation.
- Added highlevel documentation on how workspace security is suppose to
  work.
- Added tests for WorkspaceSharingPolicyPlugin since it is doing more now.
- Removed configuration for AuthzRealm and MetacardAttributeSecurityPolicyPlugin because
  they are no longer required.
- Admins now have complete control of workspaces.

djblue committed:
Added back prevenDefault to prevent username/password from being appended to url.

andrewkfiedler committed:
Rewrite cesium shape drawing for better WKT Support / UX
 - All cesium shapes are now made up of billboard points and polylines.  This greatly simplifies maintenance and support, since all other shapes can be composed of these two base shapes.
 - Updates cesium to the latest version (1.17 to 1.24).
 - Adds a shapes.js file that abstracts adding shapes to cesium.  As a result, geometry.view and geometry.collection.view don't need to directly know about cesium, and thus might be able to be used for both cesium and openlayers in the future.
 - Adds an indication of clickability to shapes that are drawn in Cesium.  As a result, hovering over a clickable shape will turn the cursor into a pointer.  Given the limitations of how Cesium, this was accomplished through ScreenSpaceEventHandler.
 - Updates panning to a result to make sure the result is completely visible.  If it isn't, the camera will adjust to make sure the entire result geometry can fit on screen (it will zoom out if necessary, but it will never zoom in).
 - Moves settings for the pageable result collection a to more visible / maintainable place.

andrewkfiedler committed:
Improve shape rendering performance
 - Updated cesium.controller to update the collection of it's geometry.collection view rather than create an entirely new view.
 - Moved call for render from cesium.controller to geometry.collection.
 - Debounced the render method for geometry.collection.

andrewkfiedler committed:
Fix jshint issue

djblue committed:
Added read-only default attributes.

djblue committed:
Updated read-only default attributes.

- Admins can specify a regular expression to mark attributes as read-only.

djblue committed:
Fixed login on <enter> press.

Adding back `e.preventDefault()` to the login function broke the login on
<enter> because it was being called without an event. Fixed by passing
event through to login function.

djblue committed:
Added an attribute for cql query results to determine if a metacard resource is local.

andrewkfiedler committed:
Rewrites clustering for cesium, adds convex hull
- Rewrites the clustering logic for cesium. In short, it gathers the geometry for each result, calculates the center point of the geometry, finds whether or not it is occluded by the horizon, and then determines the window position.  These window positions are then fed to a DBSCAN algorithm (set to cluster things within 44 pixels of one another).  The results from that are then used to drive the clustering.
- Removed old cesium toolbar css that was pushing the toolbar down.
- Moved logic general to all cesium.controllers to be within the cesium.controller rather than the cesium.view.
- Removed preloading of some images into the cesium billboard collection.
- Added a utility for converting between common cesium coordinate systems.
- Updated the cesium map to be more context friendly in terms of the user’s pointer.  By default, it is now the grabbing pointer.  When drawing it becomes a crosshair pointer.  
- Adds computation of the convex hull for each cluster.  When hovering on a cluster, this polygon is shown to the user.  Imagine it as a rubber band stretched around the outside of all the geometry that makes up each cluster.  In other words, it’s the bounds of what makes up a cluster.  
- Larger svgs are now used for the cluster markers.  
- Clusters can now be clicked on, which will selected every result within the cluster.  
- Clusters now indicate if all, some, or none of the results within the cluster are selected.  Black text with a black outline denotes full selection, white text with a black outline denotes partial selection, and white text with a white outline denotes no selection.  
- Updated the icon for enabling clustering to show when it is and is not in effect.
- Clustering is now performed while the camera is moving (instead of only after the camera stops moving) to provide a more fluid user experience.  
- Added contextual help to the clustering icon on the map.  
- Added an eyeOffset to points on the billboard to make them appear in front of polygons (for the most part it works), which is the method recommended by the Cesium developers.  Unfortunately, it only appears to take effect in 3D mode.  However, it works fairly well for making billboards appear on top of other geometries.  

djblue committed:
Display if a remote resource is cached locally.

djblue committed:
Added contextual help for cached badge to improve usability.

rzwiefel committed:
Removed unnecessary null checks with optionals

djblue committed:
Added a quick fix for persisting query sources.

djblue committed:
Fix setting the name of a query from settings tab.

- Setting the name through the settings tab would set the title as an
  array of strings, instead of a string.

djblue committed:
Updated query parse so that server saved queries will display correctly after refresh.

andrewkfiedler committed:
Update imagery provider types that had been moved within the Cesium API
 - Two of the imagery provider types had been moved in Cesium API, so they were no longer working.

andrewkfiedler committed:
Move cesium and openlayers into maps folder to prep for common interface

djblue committed:
Updated components.less

djblue committed:
Security updates:

- Moved to using taxonomy for attributes over custom workspace attributes.
- Removed embeded sharing metacard.
- Added access groups/individuals to workspaces.
- Added itest for adding/removing permissions to a workspace.
- Updated ui to use new security attributes.
- Fixed admin mapping so it gets mapped correctly.
- Updated policy plugin to prevent return of empty permissions.
- Updated workspace transformer for easier maintenance.
- Updated unit tests.

andrewkfiedler committed:
Rewrite maps to use common interface, add clustering, fix WKT shapes
 - Both maps now implement the same interface (map.js), so they can utilize anything in the visualization map folder.  This includes the clustering logic, the geometry view logic, and even the templates for the map markers.
 - Because Openlayers now fullfills the contract, it gets the benefits of clustering and full support of WKT shapes.
 - Updates how the map pans / zooms to a result in Openlayers.  It now works the same way we've implemented it in Cesium.  By default, selecting a result will cause the camera to pan to the geometry.  If the full geometry isn't visible, the camera will zoom out to fit it.  However, the camera will never zoom in.
 - Fixes issues with the gazetteer / geocoder.  It wasn't correctly going to the extents that were being found.  It will now work similar to the implementation in Cesium.  Once the extents are found by the service, the camera will pan and zoom to fit those extents.
 - For now, I added the button for clustering in Openlayers onto the template / view for the geocoder.  It shouldn't go there, but we'll be rewriting the maps to use a common toolbar soon, and it allowed me to quickly test out the clustering feature in Openlayers.

rzwiefel committed:
Fixed reducing immutable collections issue

andrewkfiedler committed:
Fix sticky footer on query editor views
 - When we added the form wrap around the query editors (in order to enable autocomplete), we failed to update the css to account for the new element.  As a result, footers weren't sticking properly.  The form element now takes up 100% height, so sticky footers once again work.

andrewkfiedler committed:
Add metacard validation to quality tab
 - Previously we were only showing attribute validation.  Now we will show both metacard and attribute validation issues.
 - I made a best attempt at turning the current duplicate message into a more useful display by turning ids into links for the user.  In the future, we should come up with a standard way to link inside messages, so we won't need fragile / brittle code here.

andrewkfiedler committed:
Turn filtering on for validation based enums, attribute sorting list

andrewkfiedler committed:
Update a few media queries to better suit smaller screen sizes
 - Updates the variable for medium screen size to 1440px, small screen size to 940px.
 - Updates the query / metacard and visualization panels to stack at lower screen sizes.
 - Turns off some of the sticky headers / footers in the views at lower screen sizes.
 - Adds help hints to the result items, so users can figure out what each value maps to.
 - Adds css to completely hide hints when help isn't shown.
 - Adds a limit to dropdown companion sizes to better suit small screens.

djblue committed:
Clean up the workspace transformer.

djblue committed:
Added more catalog-ui itests.

djblue committed:
Updated UI to use new owner key.

andrewkfiedler committed:
Add guard for results without geometry to openlayers panning
 - Previously, if a result without geometry was selected, the openlayers UI would throw an error because it would try to pan to it.

djblue committed:
Moved to using string constants in itest.

djblue committed:
Added sortOrder and sortField to query metacard attributes.

andrewkfiedler committed:
Add query settings to basic and advanced views, move title editing to outside
 - The query settings (sort and federation) now show up under the cql settings for the query on both the basic and advanced views.
 - The query title is now editable from the actual title bar, similar to how the workspace title is edited.

andrewkfiedler committed:
Update input enum to handle false values

andrewkfiedler committed:
Update dropdown styling

andrewkfiedler committed:
Fix issue with alert settings
 - Persistance was being set to a string value instead of a boolean, causing some issues with hiding and showing of further settings.

andrewkfiedler committed:
Add cursor context to workspace title
 - Adds pointer cursor to workspace title to indicate it's editable.

andrewkfiedler committed:
Fix routing for duplicates in metacard quality view
 - The previous method would destroy query params in the URL, so routing from a 2D map would cause a reload.

rzwiefel committed:
Added revert endpoint code, fixed compile errors

rzwiefel committed:
Fixed recent uploads endpoint not returning results

djblue committed:
Updated test coverage numbers.

djblue committed:
Use BASIC_METACARD directly to prevent attribute conflicts.

andrewkfiedler committed:
Fix issue with enter causing accidental button presses within forms
 - See http://stackoverflow.com/questions/8294465/making-enter-key-on-an-html-form-submit-instead-of-activating-button for more details.
 - This became an issue after we wrapped the query-basic and query-advanced views with forms to enable autocomplete.
 - This appears to be the minimal set of changes that will fix the issue for now.  I attempted to add "type='button'" to all buttons which weren't of type "submit", however it appeared to disable autocomplete when I did so.  In particular, adding "type='button'" to the edit and cancel buttons appear to cause autocomplete to stop working in Chrome, but not Firefox.  We should investigate more, as it appears there are going to be cross browser inconsistencies when it comes to this sort of behavior.

rzwiefel committed:
Fixed endpoints not returning correct content type

andrewkfiedler committed:
Add button behavior / menu-navigation decorator
 - Adds button behavior to use with views that should act similar to buttons.  It emulates a click when the user presses the space bar or enter key.  It also adds a tabindex so the element can be tabbed to.
 - Adds menu navigation decorator to add menu navigation with arrow keys / selection on enter key to various views.
 - Adds css for focus.  The outline will fade out after 1 second, so as not to be disruptive, but to still allow users to see where the focus is upon a focus change.
 - Updated query-select to be an extension of the query-item collection.
 - Updated a few custom dropdowns to use the simple dropdown.

andrewkfiedler committed:
Add table visualization
 - Adds table visualization.  It has sticky headers.  Users can reorder columns, hide/show columns, and sort by columns (if they are sortable).  Column order and hide/show preferences are persisted.  Changes to them are persisted in such a way as to preserve ordering and hide/show preferences for attributes not currently in the result set.  Users can also select (click, ctrl click, shift click) as they would on the result list to add results to the current selection.  Links are represented as anchor tags for ease of going to their destinations, and thumbnails are shown as img tags.  In order to avoid long urls bloating the table width, urls are shown as simply their property name (such as resource-download-url).
 - Moves some logic out of the result-selector.view and into the paging.view.  Particularly the setting of the active results.
 - Updates each of the selection interfaces to add an activeSearchResultsAttributes, so the calculation doesn't need to be repeated all over the place.
 - Updates highlighting of active search results to be done within result-item.view instead of result-selector.view, similar to how they are highlighted in the new row.view.  This will be more efficient and keep highlighting better in sync with selection changes.
 - Adds two new preferences to the User model, columnHide and columnOrder.  These are specifically for the table visualization.
 - Slight update to the css for neutral buttons (transparent instead of grey background).
 - Add dependency on sortablejs to package.json.  It is used for the table-rearrange.view.
 - Adds Marionette.Region.js.  This is a backport of https://github.com/marionettejs/backbone.marionette/pull/2625 to allow replacement of a region's el.

bdeining committed:
Added try/catch for better logging when the terrain provider JSON parsing fails
Fixed bug where the configuration isn't updated when the terrain provider is null

Commit:
Limiting the use of the "cache" source to only return results from sources selected for the query

bdeining committed:
Add Configuration to overwrite the Scheduling Frequency List for queries

Commit:
Corrected blueprint to inject XmlInputTransformer

andrewkfiedler committed:
Refactor polling frequency enum construction, update to use moment, reformat
 - Updated the construction of the label to use momentjs.
 - Updated enum construction to be a reduce, moved sorting to happen before the 'Never' / false option is introduced.
 - Update polling frequency enum to be generated only once, reformat file

andrewkfiedler committed:
Update result selector menu actions to restrict line wrap
 - On small screens, it's possible for the actions to start wrapping to another line.  In the future, we should handle this more gracefully by collapsing the menu actions similar to how we collapse tabs.

andrewkfiedler committed:
Update geometry views to check for selection upon creation
 - Previously they were only reacting to selection changes, so they would fail to appear selected if they weren't around before the result was selected (such as when switching between visualizations).

andrewkfiedler committed:
Update openlayers toolbar to attach to correct element
 - When multiple openlayers maps were open, the toolbar would get incorrectly attached to other instances.

andrewkfiedler committed:
Update cluster selection matching to be more performant, throttle tab resize handler
 - Updates the cluster selection check to loop over the smaller set of models (the cluster or the selection).
 - Update the tab resize handler to be throttled.

bdeining committed:
Fixed NaN on ingest modal

andrewkfiedler committed:
Rewrite association view to support graph view, two way associations
 - Users can now filter associations by all, by what is associated to the current metacard, and by what the current metacard is associated two.  Clicking through associations (routing) works on both the list and the graph view (when not in edit mode).  Both the graph and list view are editable (and synced), so users can go back and forth editing in whatever view is more comfortable for them.
 - Filtering doesn't hide the associations completely, instead it lowers their opacity to allow better awareness.  When filtering and in edit mode, only associations that apply to the filter can be edited, removed, and added.
 - Both filter and display type are persisted for the session.  It may be worth considering moving these to be cross session preferences.

andrewkfiedler committed:
Add ability to hide attributes through admin configuration
 - Similar to the read only attributes, it utilizes regex.  Moved the matching method into the properties object, added a method for checking isHidden and isReadOnly.
 - Updated details view and table view to take into account hidden attributes.
 - Note that attributes will only be hidden from the details and the table view.  This is to avoid possible clashes between the result preview list or the summary show list, and to prevent possible oddities if an admin tries to hide something like thumbnail, title, created, or modified.
 - Updated details view to sort attributes before displaying them.

andrewkfiedler committed:
Add ability to route to workspace metacards to see workspace details
 - Add an action to the workspace menu to allow routing to the workspace metacard.
 - Update routing to metacards to allow routing to any tag type.
 - Update initial fetch of Sources to be synchronous.

rzwiefel committed:
Changed Associations to support two way associations

rzwiefel committed:
fixed issue with commented out frontend plugin

rzwiefel committed:
Minor cleanup of Metacard Application code that is not used anymore

rzwiefel committed:
Resolved build issues with findbugs and checkstyle

andrewkfiedler committed:
Remove html injection opportunity
 - Since users can edit metacards, they could possibly inject html into link and thumbnail fields.  Added an escapeHTML method to Common for ease of use in the future.

Commit:
Updated standing query to use metacard.created and metacard.modified dates.

rzwiefel committed:
Fixed a key that is now incorrect because of taxonomy

bdeining committed:
Removed metadata filter in metacard view
Added default hidden attributes and removed metacard and validation filters

andrewkfiedler committed:
Add metacard overwrite view
 - Adds a metacard overwrite view so that users can upload new content for a metacard.
 - Upon a successful overwrite, the metacard is requeried and replaced in all relevant collections (see ResultUtils).
 - If the user navigates away from the view, the upload continues.
 - Overwrites are tracked, so users can leave and come back to an overwrite in progress.  Once an overwrite finishes, they can start a new one by clicking the back arrow.
 - For the time being users can still take other actions on the metacard while it's being overwritten.  It may be worth blocking all actions on metacards in the future while they're in the process of being overwritten.
 - Since overwrites are tracked, we could also send an alert in the future.

andrewkfiedler committed:
Update dropdown icons to be more consistent

andrewkfiedler committed:
Rename grid display choice to gallery

andrewkfiedler committed:
Add new ingest view / route

andrewkfiedler committed:
Add upload routing

andrewkfiedler committed:
Update terminology for association filtering, fix issue with graph view add

andrewkfiedler committed:
Deselect results upon workspace change

andrewkfiedler committed:
Add click event to dropzone text
 - Previously, clicking the dropzone text wasn't triggering file adding

andrewkfiedler committed:
Update preference saving to not parse responses

andrewkfiedler committed:
Add debounce to table header sorting
 - Since this is a potentially expensive operation, only execute when the user has stopped clicking for at least 500 ms.

andrewkfiedler committed:
Fix issue with query sources being modified upon search
 - Updated to use a copy of the searchdata when executing the search.

andrewkfiedler committed:
Notify users when a search is attempted without sources

andrewkfiedler committed:
Fix firefox float issue

andrewkfiedler committed:
Fix issue with querying without sources

andrewkfiedler committed:
Update list styling to be more consistent cross browser
 - Switched list styling to use borders instead of box shadow.  Box shadow ended up being too variant across browsers, and we can achieve much the same effect with borders.
 - Added table support for list styling.
 - Updated table to not collapse borders.
 - Added list highlighting to uploads and alerts views.
 - Fixed issue with workspace item margins.
 - Updated duplicate group styling to be more compact, readable.

andrewkfiedler committed:
Update navigation bar to more gracefully handle extreme scenarios
 - Sometimes the middle nav is dynamic (including titles), which means it could be inordinately large.  For now, we'll handle this by setting a max size on the middle nav and setting overflow in the horizontal direction.  In the future, the menu-bars that get placed in the middle nav should take this into account and gracefully degrade.

rzwiefel committed:
DDF-2520: Versioning Performance improvements

andrewkfiedler committed:
Refactor typography, add user settings area, add zooming feature
 - Refactors the typography to be more consistent across browsers.  Base font is now 'Open Sans Light', with the bold font being 'Open Sans'.  This results in better readability.
 - Adds user settings area to contain all the various settings that users now have.  This includes Theme (zoom for now, colors in the future), Alerts (persistance, expiration), Map (layers).  The user settings area is in a slideout.  Alerts and Map preferences have been removed from the view menu, which now only contains the visualization switching.
 - Adds user ability to control zoom.  It operates in much the same way as the browser controlled zoom.  It will even affect media queries, although I had to update how media queries work (and roll our own) in order to overcome the limitations of media queries (they don't quite work with rem units, in order to avoid infinite loops).  In order to allow zooming, I've replaced almost all references to px units with rem units.  I've also added some new less variables to use, such as @minimumFontSize, @largeFontSize, @minimumSpacing, @largeSpacing, etc.
 - In order to play nicely with the zoom, I've replaced the existing badging.  Instead of displaying a number, the icons have slightly decreased opacity.  If they have items, they gain full opacity.

andrewkfiedler committed:
Fixes issue with map layer preferences loading
 - Previously it would only work by chance, or if the user had already loaded the map as their visualization.

Commit:
Catalog UI bundle shutdown / startup cleanly. Resolved ConcurrentModificationException.

andrewkfiedler committed:
Refactor layer preferences to remove map, checkboxes, and apply
 - Removed apply and updated saving to be automatic.  As such, the map is no longer necessary (you can watch the main map update in real time).
 - Removed checkboxes in favor of clicking on the layer name.
 - Added opacity indication of whether or not a layer is shown.
 - Upgraded to latest cesium in order to pull in https://github.com/AnalyticalGraphicsInc/cesium/pull/4258, which had been causing issues with hiding and showing of layers too quickly (cesium would crash).

andrewkfiedler committed:
Fix issue with panning to results in Cesium when not in SCENE3D
 - The method for getRectangleCameraCoordinates appears to be broken in Cesium when in modes that aren't SCENE3D.  However, simply using the rectangle appears to work in those modes.  Do note however, that simply using the rectangle in SCENE3D will not work.  In some cases, it will be on the opposite side of the globe of where it needs to be.  As a result, we need the extra method (getRectangleCameraCoordinates), which thankfully works properly in SCENE3D.

andrewkfiedler committed:
Move fonts into css directory
 - They were being referenced from within the css directory, so the path wouldn't resolve correctly previously.

andrewkfiedler committed:
Update upload batch model to calculate total percentage rather than use dropzone's
 - Dropzone's total percentage doesn't appear to take into account all files, only the files that are currently uploading.  This can be misleading when uploading many files, and only uploading two at a time in parallel.  I went ahead and replaced the calculation with one that takes into account all the files.

andrewkfiedler committed:
Add indication of interrupted upload batches
 - If an upload batch was interrupted, this is now displayed in the summary by both highlighting and a warning icon.

andrewkfiedler committed:
Remove extra processing of date value, fix handling of empty values

andrewkfiedler committed:
Update UploadBatches to save as soon as possible after finishing
 - Most saves are throttled to 1 second, but when an upload finishes we want to save as soon as possible to avoid erroneous interruption states (saying an upload was interrupted even though it finished).  Still, given the nature of saving, it's impossible to completely rule out erroneous states.  This should at least close the gap a bit.

andrewkfiedler committed:
Fix overflow issue in workspace menu, cursor on workspace template items

Commit:
Fixed "Archive items(s)" typo

rzwiefel committed:
Refactored Patch endpoint and fixed date parsing

 - Refactored some unsightly code in the metacard patch logic

 - Fixed date parsing because boon is trying to autoparse date and the
   format ends up not being iso8601 after its done.

djblue committed:
Made system user configurable.

djblue committed:
DDF-2539: Fixed log out issue on windows.

rzwiefel committed:
Fixed rebase issues

andrewkfiedler committed:
Refactor navigation bar, notifications, & sign in.  Add navigator. Update branding.
 - Refactors navigation bar, reducing the number of icons.  Change sign in to guest with icon indicator.
 - Updates branding to add on a product name (or app name if you will).  Regardless of branding, product name stays the same and is tacked onto the end.
 - Updates application icon to feed icon.
 - Removes tasks and old notifications in favor of the newer upload notifications and alerts.
 - Consolidates notifications in one area.
 - Changes user facing text to use "notification" word instead of "alert" or "schedule".
 - Refactors sign in view.
 - Adds error message for when a user previously signed in gets signed out ("Guests cannot save preferences" was previously all we would say).
 - Updates how routing to a nonexistant XXXX is handled.  Now users are shown the navigator and prompted to choose a new route.
 - Product name is 'Catalog' for the time being.  Product icon is 'fa-feed' for the time being.
 - Since notifications are now consolidated, expiration dates / persistence applies to both upload notifications and search notifications.
 - The notifications view groups notifications by day, up until it hits things older than a week.  Notifications older than a week are all in one group called 'Older'.  Groups can be cleared all at once, in addition to allowing individual notification clearing.
 - The navigator view will show the most recent workspace (if there is one) and the most recent metacard (if there is one).  There is a placeholder for "other apps" and "about".

andrewkfiedler committed:
Fix overlapping links on actions view, allow scrolling on view
 - Links were being set to a fixed height, which caused overlapping (since overflow is visible by default).  Now links are allowed to wrap and height is auto.
 - Added overflow auto so the view can scroll if need be.
 - Updated map actions view to be more consistent with other views (now it has it's own css / tag).
 - General updates to actions css.

rzwiefel committed:
Added an enumerations/attribute endpoint and changed old url
  to /enumerations/metacardtype/:type

andrewkfiedler committed:
Update Match Types dropdown to handle 'metadata-content-type' deprecation
 - The dropdown will now search both 'metadata-content-type' and 'datatype'.  The dropdown combines values possible for both.
 - The datatype enum is fetched in addition to the other metacard definition enums.

andrewkfiedler committed:
Update time range search to include metacard.created and metacard.modified, update advanced to basic conversion
 - Updated the basic query view to include metacard.created and metacard.modified in time range searches.
 - Updated how conversions to basic view from the advanced view were done when dealing with time ranges.  Now only anyDate will convert, and only if all the filters for the dates match.

andrewkfiedler committed:
Fix issue with nested query conversion in query basic view

andrewkfiedler committed:
Add class to downconverted query views

andrewkfiedler committed:
Remove cometd and cometd related files

rzwiefel committed:
Added a localcatalogid endpoint

andrewkfiedler committed:
Delete unused files

andrewkfiedler committed:
Fix issue with non-multivalue fields having no value
 - The histogram attribute selector starts off without a value, which caused it to be hidden and unchangeable.

andrewkfiedler committed:
Fix overflow issue on html / body

andrewkfiedler committed:
Fix issue with unfederated searches appearing to not finish
 - Uses the new localcatalogid endpoint to search the proper source instead of using 'local'.
 - Previously, when the local search came back, it would not have the source id 'local', which caused the spinner to spin indefinitely.

andrewkfiedler committed:
Fix long title text overflow for association links

andrewkfiedler committed:
Fix long title overflow issue in navigator

andrewkfiedler committed:
Add icon to indicate metacard tag to result preview
 - Adds indicator of whether a result is workspace, history item, resource, or deleted.

andrewkfiedler committed:
Fix autocomplete in Chrome
 - Chrome recently updated, causing the old method to only work once per page load.

rzwiefel committed:
DDF-2520: refactored storage update to deal with missing subject

andrewkfiedler committed:
Fix USNG issue
 - The values of the actual bounding box drawn were being used instead of the USNG values.
 - Updated switching from USNG back to Lat / Lon to make it preserve the Lat / Lon corresponding to the USNG location.

andrewkfiedler committed:
Add support for different metacard types to various views
 - Updates views to support the four base metacard types: resource, workspace, revision, deleted

andrewkfiedler committed:
Remove routing to recent uploads, remove unnecessary toolbar dropdowns

andrewkfiedler committed:
Remove view from content toolbar, move visualization prefs to user-settings slideout
 - Users can now also choose 2dmap or 3dmap as a preference, rather than using url params

andrewkfiedler committed:
Refactor blacklist, move to user settings

andrewkfiedler committed:
Remove old restore view

andrewkfiedler committed:
Update scrolling in user-settings

andrewkfiedler committed:
Add unseen notification indication

bdeining committed:
Refactored Email Notification configuration and scheduler (#26)

* Refactored Email Notification configuration and scheduler

rzwiefel committed:
Updated History endpoints to work with new versioning changes

rzwiefel committed:
Fixed findbugs issues within MetacardApplication

rzwiefel committed:
Fixed more findbugs issues

rzwiefel committed:
Fixed workspacequeryservice unit tests

rzwiefel committed:
Fixed checkstyle issues

rzwiefel committed:
Fixes for catalog ui itests

rzwiefel committed:
Added support for associations on non-resource types

andrewkfiedler committed:
Remove content toolbar, add to workspace-interactions menu
 - Content toolbar now completely gone.
 - Added functionality from file menu to workspace interactions menu.
 - Navbar now more consistent across pages, and simpler.
 - Added placeholder for workspace title and search title.

andrewkfiedler committed:
Comment out "About" and "Other Apps"
 - Until we have someone to route the user, these will be hidden

andrewkfiedler committed:
Increase timeframe for resizing ingest details

rzwiefel committed:
Fixed injected attributes not being returned in metacardtype endpoint

rzwiefel committed:
Change byte[] persist type to String

rzwiefel committed:
Updated User application to store preferences as binary instead of string

rzwiefel committed:
Fixed findbugs issue in UserApplication

rzwiefel committed:
DDF-2567: correct way of dealing with solr

rzwiefel committed:
Fixes persistence store errors

bdeining committed:
Fixed metacard-tag issue

rzwiefel committed:
Fixed issue with reverting
 - Catalog overwrite does not add a resource URI onto the metacard

andrewkfiedler committed:
Add "Search for Deleted" to query interactions
 - This allows users to more easily find deleted items that match a particularly search, since they won't have to go back and modify the actual search.

andrewkfiedler committed:
Add ability to restore a single metacard
 - Archive tab no longer hidden for singularly selected metacards.
 - Renamed query interaction "Search for Deleted" to "Search Archived".
 - Update refresh data call in Metacard to be tag agnostic.
 - Update metacard(s) tabs to refresh available tabs / content upon a refresh data call on a selected metacard.

andrewkfiedler committed:
Update basic type searching to use "ILIKE" instead of "="

andrewkfiedler committed:
Remove ability to edit remote metacards
 - Removes ability to edit remote metacards, since that is not supported.
 - Remote metacard tabs are limited to: Summary, Details, Actions
 - Remove remote metacards from possible association values

andrewkfiedler committed:
Fix issue with inability to draw on 2d map (openlayers)
 - The check for whether or not the map was 2d had to be updated to not use the url param.

andrewkfiedler committed:
Update histogram attribute selector to use attribute aliases

andrewkfiedler committed:
Add support for wildcard matching to local results filter

andrewkfiedler committed:
Scale back animated properties for slideout to be more performant
 - Removes the transform on the main content to allow better performance.  Blur updated to happen immediately.
 - Content isn't emptied from slideouts until the the end of their hide animation now.

rzwiefel committed:
Fixed Associations

 - Does not allow restricted metacard tags to be searched now. (revision, and deleted)

rzwiefel committed:
Fixed reverting update issue and added logging

bdeining committed:
Removed Result Preview Attributes when they are not set

andrewkfiedler committed:
Update routing / results to handle high latency, fix issue with datepicker
 - Updates all routing to handle high latency.
 - Updates results view to handle high latency.  Users are asked to merge (if time since last merge exceeds 250).  Indication of a search in progress added to results view.
 - Datepicker sometimes did not properly close.  Now it will be destroyed when the view it belongs to is destroyed.
 - Updates openlayers font size to be 16px for the time being.  The toolbar doesn't not correctly resize with zoom %.

bdeining committed:
Made the Email Notification Query interval configurable with a default of 24 hours (#32)

* Made the Email Notification Query interval configurable with a default of 24 hours

* Code Review

andrewkfiedler committed:
Fix issue with result sorting, allow automerge of unviewed results
 - Update to sort after each result set is added to results.
 - Update to allow automerge of unviewed results.  Shortened threshold for automerge during view to 16 ms.  Throttled merge notification to 2.5 seconds rather than 250ms.

bdeining committed:
Fixed Remote Metacard Assertion (#36)

andrewkfiedler committed:
Add error handling for workspace deletion

andrewkfiedler committed:
Add polling to persisted query attributes, fix dirty workspace check
 - Polling is now persisted for cloud based workspaces.
 - Dirty workspace check now works when logged in.

andrewkfiedler committed:
Fix issue with username overflowing

andrewkfiedler committed:
Make map actions appear clickable

andrewkfiedler committed:
Update archiving an item to immediately tag results as deleted
 - On high latency systems, the refresh can take too long (allowing users to possibly start taking inappropriate actions on the archived item).  This will update the tags immediately to ensure it's treated appropriately.

andrewkfiedler committed:
Update association dropdown to show full result set
 - Updates association dropdown to show full result set that the result belongs too (previously it only showed the current page).
 - Fix issue with being able to access improper tabs as a remote metacard.
 - Fix issue with not setting currentlyViewed on a result set.

andrewkfiedler committed:
Fix location input to respond to zoom level

andrewkfiedler committed:
Enable custom error handling for validation calls

Commit:
Udated workspace policy to allow metacard owners access permissions

bdeining committed:
Fixed catalog-search-ui to shutdown properly

andrewkfiedler committed:
Updates default user visualization from map to 3dmap

andrewkfiedler committed:
Update map views to display current location filters
 - When viewing results, relevant query location filters are now displayed on the map.
 - When leaving the results tab, results are now removed from the map.
 - The global location filter is always displayed on the map.

andrewkfiedler committed:
Add federation to persisted query attributes

andrewkfiedler committed:
Remove blur on slideout, add dropshadow
 - Removes blur from content on slideout to cater to machines without hardware acceleration.
 - Adds dropshadow to slideout content.

andrewkfiedler committed:
Add support for systems without datatype enum

rzwiefel committed:
No longer do Enterprise searches for any action on metacard

 - Enumerations endpoint will now return empty map when it cannot find
   the metacard type specified

andrewkfiedler committed:
Update query response parsing to add new metacard definitions
 - Update query response parsing to add previously unknown definitions.
 - Update sorted types list to be case insensitive.
 - Update instances of anyText '%' to '*'

andrewkfiedler committed:
Update getValidation to ignore remote metacards

andrewkfiedler committed:
Remove unused code

andrewkfiedler committed:
Remove unused files

andrewkfiedler committed:
Remove unused code / files

andrewkfiedler committed:
Update geometry based filters to always use anyGeo property
 - Geometry attributes are stripped out before the metacards are sent to the UI, so they are only left with the geometry field.  As such, any search on a geometry attribute will now be updated to use the anyGeo property which will then search that geometry field.

andrewkfiedler committed:
Add error handling to metacard edit views

rzwiefel committed:
Fixed the metacard update error message to be more generic (#40)

andrewkfiedler committed:
Update local geometry filtering to use intersection
 - Previous it was contains, which didn't match the cql endpoint.  This caused unexpected behavior for users.
 - Terraformer didn't support intersects on anything but LineString, MultiLineString, and Polygon.  I added the method 'intersects' to Metacard to add support for the other geometry types.  Terraformer also incorrectly handled MultiPolygons, so the 'intersects' method turns them into Polygons first.
 - Ideally the 'intersects' method could become a PR to Terraformer to fix / improve their existing intersects method.  Then we could remove this method and purely rely on Terraformer.
 - Added guards for when Lines or Circles are 0 or less in width / radius.

andrewkfiedler committed:
Fix incorrect result count on query feed
 - state.totalRecords was not returning the correct amount.  It's now updated to use fullCollection.length, which should always work.

andrewkfiedler committed:
Update sort list to only hide anyGeo and anyText
 - Previously it was hiding a few more, as well as any attribute that wasn't a date or string.

andrewkfiedler committed:
Fix issue with map drawings not cleaning up completely, selection during drawing
 - Updated drawing logic to add more cleanup upon destruction.
 - Updated selection of geometry to only happen when not drawing.

andrewkfiedler committed:
Update line drawing to default to 1 meter instead of 0, enforce lower bound
 - Line drawing now defaults to 1 meter for ease of use.
 - Lower bound of line width is now set to 0.000001, since the cql parser we're using can't handle anything lower than that.

andrewkfiedler committed:
Add lower bound to radius for circle drawing

andrewkfiedler committed:
Add validation to location model
 - Validation is now applied to point radius, line, and bbox drawing.  Point radius lat / lon are validated, as well as the radius value itself.  Line width is now validated.  BBox lat / lon values are validated.
 - Added try catch to usng code to prevent errors for special cases (outside -80 and 84 degrees).  These are still valid outside usng, so we shouldn't be preventing them.  They're also valid in usng technically, we just don't offer support yet.

andrewkfiedler committed:
Update convex hull to be transparent, remove z-index adjustment from cesium points
 - Previously it was the same color as the cluster it belonged to.  Since there isn't a "with certainty" method to make sure that points are above their convex hull in Cesium, it is now transparent.
 - Z-index adjustment has been removed from cesium points.  This was an attempt to make them appear above convex hull outlines, but it ended up causing issues at extreme zoom levels (points would disappear prematurely).

andrewkfiedler committed:
Update to hide hidden attributes more consistently, expand scope
 - Previously hidden attribute configuration was only applied to the summary and details view.  Now it's applied everywhere (filters, sorts, etc.).

rzwiefel committed:
Added missing pom dependencies

rzwiefel committed:
Removed references of catalog-ui from search-ui

rzwiefel committed:
Fixed null value enumerator hack

Commit:
Updated Catalog UI configurations
- removed blueprint default configurations
- added catalog UI config file with default configurations
- updated hidden attribute config to include workspace and query attributes

Commit:
Merge pull request #43 from codice/ui-hidden-config

Updated Catalog UI configurations

rzwiefel committed:
Changed revert endpoint path to allow security (#42)

rzwiefel committed:
Fixed workspace update failure (#41)

andrewkfiedler committed:
Add boolean / number support to query / filter
 - Adds boolean support to cql writer / reader.
 - Adds boolean input to specifically handle boolean fields.
 - Adds number input to specifically handle number fields.
 - Adds support for boolean and number searching to both query and local filtering.

andrewkfiedler committed:
Update metatype config for hidden / readonly attributes to use proper regexp

andrewkfiedler committed:
Fix issue with MGRS BBox being overridden
 - The "if" statements weren't quite right for the model validation, updated to only compare to current value if existing value isn't being updated.
 - Updated setLatLon to only be called if in latlon mode.

Commit:
bump gt-cql version number

Commit:
Merge pull request #45 from glenhein/bump-cql

bump gt-cql version number

rzwiefel committed:
Cleaned up todos and some comments (#44)

rzwiefel committed:
Cleaned up comments, variables, and javadoc

rzwiefel committed:
Removed unecessary DeleteRequestImpl change

pklinef commented:
Manually merged
